source/common/common/logger.h

```diff
@@ -430,6 +431,22 @@ template <Id id> class Loggable {
     }                                                                                              \
   } while (0)
 
+// This is to get us to pass the format check. We reference a real-world time source here.
+// I think it's not easy to mock/simulate time here as it stands today anyway.
+using t_logclock = std::chrono::steady_clock; // NOLINT
+
+#define ENVOY_LOG_PERIODIC(LEVEL, CHRONO_DURATION, ...)                                            \
+  do {                                                                                             \
+    static auto* last_hit = new std::atomic<uint64_t>();                                           \
```

Comment:
Just wondering. I can see a pattern of not checking a log `LEVEL` first. Shouldn't the actual `LEVEL` be checked before doing stuff, to avoid situations where the messages might not be logged at all? Should the stopper restart anytime the `ENVOY_LOG_*` is called, or should the `LEVEL` be rather checked before? Can a message `LEVEL` not even be logged? If so, there's a possibility of not logging a message at all - say you're frequently pushing messages (timer restarts every time you're calling `ENVOY_LOG_PERIODIC`) while only some of them are actually important enough to get published into the logs.